From 9bd221abfb198fcdf51873e91201a2e717331ae0 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Tue, 16 Sep 2014 15:03:27 +0200
Subject: config.guess is apparently not supposed to be called directly

(if you look at how it is used by generated configure scripts); this fixes the
inadvertent loss of executable flag in 3379a4d028a77de5a50e305175d80f65560d5688
"Update config.{guess,sub}"

Change-Id: I29b3467970093006c994af63cb5f5bf087831ae6

diff --git a/odk/config/configure.pl b/odk/config/configure.pl
index dde3a94..f86a4f0 100755
--- a/odk/config/configure.pl
+++ b/odk/config/configure.pl
@@ -34,7 +34,7 @@ $main::OO_MINORVERSION =~ s#[^\d]+(\d).(\d).+#$2#go;
 
 $main::OO_SDK_CONFIG_HOME= "$ENV{HOME}/$main::OO_SDK_NAME";
 
-$main::operatingSystem = `$main::sdkpath/config.guess | cut -d"-" -f3,4`;
+$main::operatingSystem = `/bin/sh $main::sdkpath/config.guess | cut -d"-" -f3,4`;
 chomp ($main::operatingSystem);
 
 $main::OO_SDK_HOME = $main::sdkpath;
diff --git a/odk/config/setsdkenv_unix.sh.in b/odk/config/setsdkenv_unix.sh.in
index b721e27..4ab9d62 100644
--- a/odk/config/setsdkenv_unix.sh.in
+++ b/odk/config/setsdkenv_unix.sh.in
@@ -89,7 +89,7 @@ then
 fi
 
 # Get the operating system.
-sdk_platform=`${OO_SDK_HOME}/config.guess | cut -d"-" -f3,4`
+sdk_platform=`/bin/sh ${OO_SDK_HOME}/config.guess | cut -d"-" -f3,4`
 
 # Set the directory name.
 programdir=program
@@ -143,7 +143,7 @@ fi
 # Set the directory name.
 case ${sdk_platform} in
   solaris*)
-    sdk_proctype=`${OO_SDK_HOME}/config.guess | cut -d"-" -f1`
+    sdk_proctype=`/bin/sh ${OO_SDK_HOME}/config.guess | cut -d"-" -f1`
     if [ "${sdk_proctype}" = "sparc" ]
     then
         directoryname=solsparc
-- 
cgit v0.10.2

